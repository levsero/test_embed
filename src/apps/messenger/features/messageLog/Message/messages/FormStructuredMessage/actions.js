import { createAsyncThunk } from '@reduxjs/toolkit'
import { getClient } from 'src/apps/messenger/suncoClient'

const submitForm = createAsyncThunk('form/submit', async ({ formId, fields, values }) => {
  const responseFields = fields.map(field => {
    return {
      type: field.type,
      name: field.name,
      label: field.label,
      [field.type]: values[field._id]
    }
  })

  const response = await getClient().sendFormResponse(responseFields, formId)

  if (Array.isArray(response?.body?.messages)) {
    return { messages: response.body.messages }
  }
})

export { submitForm }
