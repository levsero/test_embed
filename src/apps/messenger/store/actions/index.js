import { createAction } from '@reduxjs/toolkit'

export const configReceived = createAction('messenger/configReceived')
