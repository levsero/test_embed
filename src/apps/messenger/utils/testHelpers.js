import React from 'react'
import { IdManager } from '@zendeskgarden/react-selection'
import { render as rtlRender } from '@testing-library/react'
import { Provider } from 'react-redux'
import { ThemeProvider, DEFAULT_THEME } from '@zendeskgarden/react-theming'
import createStore from 'src/apps/messenger/store'
import configureMockStore from 'redux-mock-store'

const mockStore = configureMockStore()

export const render = (ui, { render, store, state, themeProps = {} } = {}) => {
  IdManager.setIdCounter(0)
  if (state) {
    store = mockStore(state)
  }
  const reduxStore = store || createStore(state)

  const renderFn = render || rtlRender
  return {
    ...renderFn(
      <Provider store={reduxStore}>
        <ThemeProvider theme={{ ...DEFAULT_THEME, ...themeProps }}>{ui}</ThemeProvider>
      </Provider>
    )
  }
}
