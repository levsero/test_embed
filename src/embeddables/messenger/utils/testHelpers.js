import { IdManager } from '@zendeskgarden/react-selection'
import { render as rtlRender } from '@testing-library/react'

export const render = (ui, { render } = {}) => {
  IdManager.setIdCounter(0)
  const renderFn = render || rtlRender
  return {
    ...renderFn(ui)
  }
}
