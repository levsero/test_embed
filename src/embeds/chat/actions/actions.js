import * as actionTypes from './action-types'

export function updateMenuVisibility(visible) {
  return {
    type: actionTypes.UPDATE_CHAT_MENU_VISIBILITY,
    payload: visible
  }
}
