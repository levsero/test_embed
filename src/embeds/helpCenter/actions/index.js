import _ from 'lodash'

import { MAXIMUM_SEARCH_RESULTS } from 'src/constants/helpCenter'
import { http } from 'service/transport'
import { location } from 'utility/globals'
import {
  getAuthToken,
  getHasWidgetShown,
  getIsAuthenticationPending
} from 'src/redux/modules/base/base-selectors'
import {
  getLastSearchTimestamp,
  getContextualHelpRequestNeeded,
  getSearchQuery
} from 'embeds/helpCenter/selectors'
import { getHasPassedAuth } from 'src/redux/modules/selectors/helpCenter-linked-selectors'
import { i18n } from 'service/i18n'
import { MAXIMUM_CONTEXTUAL_SEARCH_RESULTS } from 'src/constants/helpCenter'
import {
  SEARCH_REQUEST_SENT,
  SEARCH_REQUEST_SUCCESS,
  SEARCH_REQUEST_FAILURE,
  CONTEXTUAL_SEARCH_REQUEST_SENT,
  CONTEXTUAL_SEARCH_REQUEST_SUCCESS,
  CONTEXTUAL_SEARCH_REQUEST_FAILURE,
  ARTICLE_VIEWED,
  ORIGINAL_ARTICLE_CLICKED,
  ARTICLE_CLOSED,
  ADD_RESTRICTED_IMAGE,
  GET_ARTICLE_REQUEST_SENT,
  GET_ARTICLE_REQUEST_SUCCESS,
  GET_ARTICLE_REQUEST_FAILURE,
  SEARCH_FIELD_CHANGED,
  CONTEXTUAL_SUGGESTIONS_MANUALLY_SET
} from './action-types'
import { updateQueue } from 'src/redux/modules/base'
import { isOnHostMappedDomain } from 'utility/pages'
import {
  getSettingsHelpCenterFilter,
  getSettingsHelpCenterLocaleFallbacks
} from 'src/redux/modules/settings/settings-selectors'
const constructHelpCenterPayload = (path, query, doneFn, failFn, filter) => {
  const token = getAuthToken()
  const forceHttp = isOnHostMappedDomain() && location.protocol === 'http:'
  const queryParams = _.extend(query, filter)

  return {
    method: 'get',
    forceHttp: forceHttp,
    useHostMappingIfAvailable: isOnHostMappedDomain(),
    path,
    query: queryParams,
    authorization: token ? `Bearer ${token}` : '',
    callbacks: {
      done: doneFn,
      fail: failFn
    }
  }
}

const formatResults = response => {
  const { results = [], count = 0 } = response.body

  return {
    articles: results,
    resultsCount: count,
    locale: results.length > 0 ? results[0].locale : ''
  }
}

const preventSearchCompleteDispatch = (getState, timestamp) => {
  /* This is implemented in order to prevent slow connections from returning
     successful responses in quick succession which causes screen states to
     flicker on the front-end which causes a weird experience and could potentially
     render an incorrect state. To prevent that we only dispatch the promised callbacks
     if it's the last message recorded via its ID. */
  return timestamp !== getLastSearchTimestamp(getState())
}

export function performImageSearch(path, done) {
  http.getImage(constructHelpCenterPayload(path, null, done))

  // Temporary to stop middleware from breaking until we properly implement images
  return { type: '' }
}

export function performSearch(searchTerm, success = () => {}, fail = () => {}, localeIndex = 0) {
  return (dispatch, getState) => {
    // When localeFallbacks is defined in the zESettings object then
    // attempt the search with each locale in that array in order. Otherwise
    // try the search with no locale (injects an empty string into localeFallbacks).
    const locales = [i18n.getLocale()].concat([
      getSettingsHelpCenterLocaleFallbacks(getState()) || ['']
    ])

    if (localeIndex >= locales.length) {
      return
    }

    const query = {
      locale: locales[localeIndex],
      query: searchTerm,
      per_page: MAXIMUM_SEARCH_RESULTS,
      origin: 'web_widget'
    }

    const timestamp = Date.now()
    const path = '/api/v2/help_center/articles/embeddable_search.json'

    const successFn = response => {
      if (preventSearchCompleteDispatch(getState, timestamp)) return

      dispatch({
        type: SEARCH_REQUEST_SUCCESS,
        payload: formatResults(response)
      })

      if (response.body.count > 0 || _.isEmpty(locales)) {
        success(response)
      } else {
        dispatch(performSearch(searchTerm, success, fail, localeIndex + 1))
      }
    }

    const failFn = error => {
      if (preventSearchCompleteDispatch(getState, timestamp)) return

      dispatch({ type: SEARCH_REQUEST_FAILURE })
      fail(error)
    }

    dispatch({
      type: SEARCH_REQUEST_SENT,
      payload: { searchTerm: query.query, timestamp }
    })

    const filter = getSettingsHelpCenterFilter(getState())

    http.send(constructHelpCenterPayload(path, query, successFn, failFn, filter))
  }
}

export function contextualSearch(onDone) {
  return (dispatch, getState) => {
    const state = getState()

    if (getContextualHelpRequestNeeded(state)) {
      if (getHasPassedAuth(state)) {
        dispatch(performContextualSearch(onDone, onDone))
      } else if (getIsAuthenticationPending(state)) {
        dispatch(updateQueue({ performContextualSearch: {} }))
      }
    }
  }
}

export function performContextualSearch(done = () => {}, fail = () => {}) {
  const timestamp = Date.now()

  return (dispatch, getState) => {
    const searchQuery = getSearchQuery(getState())

    if (!searchQuery.query && !searchQuery.label_names) {
      return
    } // eslint-disable-line camelcase

    const path = '/api/v2/help_center/articles/embeddable_search.json'
    /* eslint camelcase:0 */
    let query = {
      locale: i18n.getLocale(),
      per_page: MAXIMUM_CONTEXTUAL_SEARCH_RESULTS,
      ...searchQuery
    }

    const doneFn = response => {
      if (preventSearchCompleteDispatch(getState, timestamp)) return

      dispatch({
        type: CONTEXTUAL_SEARCH_REQUEST_SUCCESS,
        payload: formatResults(response)
      })
      done(response)
    }
    const failFn = error => {
      if (preventSearchCompleteDispatch(getState, timestamp)) return

      dispatch({ type: CONTEXTUAL_SEARCH_REQUEST_FAILURE })
      fail(error)
    }

    dispatch({
      type: CONTEXTUAL_SEARCH_REQUEST_SENT,
      payload: {
        searchTerm: searchQuery.query || searchQuery.label_names,
        timestamp
      }
    })

    const filter = getSettingsHelpCenterFilter(getState())

    http.send(constructHelpCenterPayload(path, query, doneFn, failFn, filter))
  }
}

export function handleOriginalArticleClicked() {
  return {
    type: ORIGINAL_ARTICLE_CLICKED
  }
}

export function handleArticleView(article) {
  return {
    type: ARTICLE_VIEWED,
    payload: article
  }
}

export function closeCurrentArticle() {
  return { type: ARTICLE_CLOSED }
}

export function addRestrictedImage(img) {
  return {
    type: ADD_RESTRICTED_IMAGE,
    payload: img
  }
}

export function handleSearchFieldChange(value) {
  return {
    type: SEARCH_FIELD_CHANGED,
    payload: value || ''
  }
}

export function displayArticle(articleId) {
  return dispatch => {
    dispatch({ type: GET_ARTICLE_REQUEST_SENT })
    const forceHttp = isOnHostMappedDomain() && location.protocol === 'http:'

    http.get(
      {
        method: 'get',
        forceHttp: forceHttp,
        path: `/api/v2/help_center/articles/${articleId}.json`,
        useHostMappingIfAvailable: isOnHostMappedDomain(),
        callbacks: {
          done: res =>
            dispatch({
              type: GET_ARTICLE_REQUEST_SUCCESS,
              payload: res.body.article
            }),
          fail: err =>
            dispatch({
              type: GET_ARTICLE_REQUEST_FAILURE,
              payload: err.response ? err.response : err
            })
        }
      },
      false
    )
  }
}

export function setContextualSuggestionsManually(options, onDone) {
  return (dispatch, getState) => {
    dispatch({
      type: CONTEXTUAL_SUGGESTIONS_MANUALLY_SET,
      payload: options
    })

    if (getHasWidgetShown(getState())) {
      dispatch(contextualSearch(onDone))
    }
  }
}
