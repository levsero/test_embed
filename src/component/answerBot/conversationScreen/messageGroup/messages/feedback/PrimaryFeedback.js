import React from 'react'
import { useDispatch } from 'react-redux'

import useTranslate from 'src/hooks/useTranslate'
import { ButtonGroup } from 'component/button/ButtonGroup'
import PillButton from 'src/embeds/answerBot/components/PillButton'
import { sessionResolved } from 'src/embeds/answerBot/actions/sessions'
import {
  botFeedback,
  botFeedbackMessage,
  botUserMessage
} from 'src/embeds/answerBot/actions/root/bot'

import { Container } from './styles'

const PrimaryFeedback = () => {
  const translate = useTranslate()
  const dispatch = useDispatch()
  const handleYes = () => {
    dispatch(sessionResolved())
    dispatch(botUserMessage('embeddable_framework.answerBot.article.feedback.yes'))
    dispatch(botFeedbackMessage('embeddable_framework.answerBot.msg.yes_acknowledgement'))
    dispatch(botFeedbackMessage('embeddable_framework.answerBot.msg.prompt_again_after_yes'))
  }

  const handleNo = () => {
    dispatch(botUserMessage('embeddable_framework.answerBot.article.feedback.no.need_help'))
    dispatch(botFeedbackMessage('embeddable_framework.answerBot.article.feedback.no.reason.title'))
    dispatch(botFeedback('secondary'))
  }

  return (
    <Container>
      <ButtonGroup>
        <PillButton
          label={translate('embeddable_framework.answerBot.article.feedback.yes')}
          onClick={handleYes}
        />
        <PillButton
          label={translate('embeddable_framework.answerBot.article.feedback.no.need_help')}
          onClick={handleNo}
        />
      </ButtonGroup>
    </Container>
  )
}

export default PrimaryFeedback
