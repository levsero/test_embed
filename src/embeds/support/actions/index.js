import _ from 'lodash'

import * as actionTypes from './action-types'
import attachmentSender from 'src/embeds/support/utils/attachment-sender'
import { i18n } from 'service/i18n'
import { getMaxFileSize, getMaxFileCount, getValidAttachments } from 'src/embeds/support/selectors'
import formatRequestData from 'src/embeds/support/utils/requestFormatter'
import { http } from 'service/transport'
import withRateLimiting from 'utility/rateLimiting'
import { FORM_PREFILLED } from 'embeds/support/actions/action-types'
import history from 'service/history'
import routes from 'embeds/support/routes'
import { getNewSupportEmbedEnabled } from 'embeds/support/selectors'

let attachmentUploaders = {}

export const submitForm = state => ({
  type: actionTypes.SUBMITTED_FORM,
  payload: { state }
})

export const attachmentLimitExceeded = () => ({
  type: actionTypes.ATTACHMENT_LIMIT_EXCEEDED
})

export const clearLimitExceededError = () => ({
  type: actionTypes.CLEAR_LIMIT_EXCEEDED_ERROR
})

export const setActiveFormName = name => ({
  type: actionTypes.SET_ACTIVE_FORM_NAME,
  payload: { name }
})

export const clearActiveFormName = () => ({
  type: actionTypes.CLEARED_ACTIVE_FORM_NAME
})

export const setFormState = (name, newFormState) => ({
  type: actionTypes.SET_FORM_STATE,
  payload: { name, newFormState }
})

export const clearFormStates = () => ({ type: actionTypes.CLEARED_FORM_STATES })

export const clearFormState = name => ({
  type: actionTypes.CLEARED_FORM_STATE,
  payload: { name }
})

const uploadAttachmentRequest = attachment => ({
  type: actionTypes.ATTACHMENT_UPLOAD_REQUESTED,
  payload: attachment
})

const uploadAttachmentSuccess = (id, response) => {
  let payload = { id, uploading: false }

  try {
    const uploadToken = JSON.parse(response.text).upload.token

    if (uploadToken) {
      payload = { ...payload, uploadToken }
    } else {
      throw new Error()
    }
  } catch {
    payload = {
      ...payload,
      errorMessage: i18n.t('embeddable_framework.chat.attachments.error.unknown_error')
    }
  }

  return {
    type: actionTypes.ATTACHMENT_UPLOAD_SUCCEEDED,
    payload
  }
}

const uploadAttachmentFailure = (id, error) => ({
  type: actionTypes.ATTACHMENT_UPLOAD_FAILED,
  payload: {
    id,
    uploading: false,
    errorMessage:
      error.message || i18n.t('embeddable_framework.chat.attachments.error.unknown_error')
  }
})

const uploadAttachmentUpdate = (id, progress) => ({
  type: actionTypes.ATTACHMENT_UPLOAD_UPDATED,
  payload: {
    id,
    uploadProgress: progress.percent || 100
  }
})

const removeAttachment = id => ({
  type: actionTypes.ATTACHMENT_REMOVED,
  payload: { id }
})

export const clearAttachments = () => (dispatch, _getState) => {
  attachmentUploaders = {}

  dispatch({ type: actionTypes.ATTACHMENTS_CLEARED })
}

export const deleteAttachment = id => (dispatch, _getState) => {
  try {
    attachmentUploaders[id].abort()
    delete attachmentUploaders[id]
  } catch {}

  dispatch(removeAttachment(id))
}

export const uploadAttachment = file => (dispatch, getState) => {
  const maxFileSize = getMaxFileSize(getState())
  const maxSize = Math.round(maxFileSize / 1024 / 1024)
  const fileOversize = file.size >= maxFileSize
  const errorMessage = fileOversize
    ? i18n.t('embeddable_framework.submitTicket.attachments.error.size', { maxSize })
    : null
  const id = _.uniqueId()
  const fileType = file.type || 'application/octet-stream'
  const onUploadComplete = response => dispatch(uploadAttachmentSuccess(id, response))
  const onUploadFailure = error => dispatch(uploadAttachmentFailure(id, error))
  const onUploadUpdate = progress => dispatch(uploadAttachmentUpdate(id, progress))
  const attachment = {
    id,
    fileName: file.name,
    fileSize: file.size,
    fileType,
    errorMessage,
    fileUrl: null,
    uploading: !fileOversize,
    uploadProgress: 0,
    uploadToken: null
  }

  dispatch(uploadAttachmentRequest(attachment))

  if (!fileOversize) {
    attachmentUploaders[id] = attachmentSender(
      file,
      onUploadComplete,
      onUploadFailure,
      onUploadUpdate
    )
  }
}

export function submitTicket(formState, formTitle) {
  return (dispatch, getState) => {
    return new Promise((resolve, reject) => {
      const state = getState()
      const attachments = [] // Will update once https://zendesk.atlassian.net/browse/EWW-992 is done
      const params = formatRequestData(state, formState, attachments, formTitle)

      const payload = {
        method: 'post',
        path: '/api/v2/requests',
        params: params,
        callbacks: {
          done(res) {
            dispatch({
              type: actionTypes.TICKET_SUBMISSION_REQUEST_SUCCESS,
              payload: JSON.parse(res.text)
            })
            if (getNewSupportEmbedEnabled(state)) {
              history.replace(routes.success())
            }
            resolve()
          },
          fail(err) {
            dispatch({
              type: actionTypes.TICKET_SUBMISSION_REQUEST_FAILURE,
              payload: err.timeout
                ? i18n.t('embeddable_framework.submitTicket.notify.message.timeout')
                : i18n.t('embeddable_framework.submitTicket.notify.message.error')
            })
            reject()
          }
        }
      }

      dispatch({
        type: actionTypes.TICKET_SUBMISSION_REQUEST_SENT
      })

      withRateLimiting(http.send, payload, 'TICKET_SUBMISSION_REQUEST', () => {
        dispatch({
          type: actionTypes.TICKET_SUBMISSION_REQUEST_FAILURE,
          payload: i18n.t('embeddable_framework.common.error.form_submission_disabled')
        })
      })
    })
  }
}

export const uploadAttachedFiles = files => (dispatch, getState) => {
  const state = getState()
  const maxFileCount = getMaxFileCount(state)
  const validAttachments = getValidAttachments(state)

  const numAttachments = validAttachments.length
  const numFilesToAdd = maxFileCount - numAttachments

  _.slice(files, 0, numFilesToAdd).forEach(file => {
    dispatch(uploadAttachment(file))
  })

  if (numAttachments + files.length > maxFileCount) {
    dispatch(attachmentLimitExceeded())
  }
}

export const formPrefilled = (formId, prefillId) => ({
  type: FORM_PREFILLED,
  payload: {
    formId,
    prefillId
  }
})
