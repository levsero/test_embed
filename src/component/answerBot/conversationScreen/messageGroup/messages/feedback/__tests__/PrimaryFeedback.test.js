import _ from 'lodash'
import { fireEvent } from '@testing-library/react'
import React from 'react'
import { render } from 'src/util/testHelpers'
import * as botActions from 'src/redux/modules/answerBot/root/actions/bot'
import * as sessionActions from 'src/redux/modules/answerBot/sessions/actions/session-resolved'

import PrimaryFeedback from '../PrimaryFeedback'

const renderComponent = (props = {}) => {
  const componentProps = _.merge({}, props)

  jest.spyOn(botActions, 'botUserMessage')
  jest.spyOn(botActions, 'botFeedbackMessage')
  jest.spyOn(botActions, 'botFeedback')
  jest.spyOn(sessionActions, 'sessionResolved').mockImplementation(() => () => {})

  return render(<PrimaryFeedback {...componentProps} />)
}

test('renders the expected classes', () => {
  const { container } = renderComponent()

  expect(container).toMatchSnapshot()
})

describe('on yes click', () => {
  it('fires the expected actions', () => {
    const result = renderComponent()
    const { getByText } = result

    fireEvent.click(getByText('Yes'))
    expect(sessionActions.sessionResolved).toHaveBeenCalled()
    expect(botActions.botFeedbackMessage).toHaveBeenNthCalledWith(
      1,
      'embeddable_framework.answerBot.msg.yes_acknowledgement'
    )
    expect(botActions.botFeedbackMessage).toHaveBeenNthCalledWith(
      2,
      'embeddable_framework.answerBot.msg.prompt_again_after_yes'
    )
    expect(botActions.botUserMessage).toHaveBeenCalledWith(
      'embeddable_framework.answerBot.article.feedback.yes'
    )
  })
})

describe('on no click', () => {
  it('fires the expected actions', () => {
    const { getByText } = renderComponent()

    fireEvent.click(getByText('No, I need help'))
    expect(botActions.botFeedback).toHaveBeenCalledWith('secondary')
    expect(botActions.botFeedbackMessage).toHaveBeenCalledWith(
      'embeddable_framework.answerBot.article.feedback.no.reason.title'
    )
    expect(botActions.botUserMessage).toHaveBeenCalledWith(
      'embeddable_framework.answerBot.article.feedback.no.need_help'
    )
  })
})
