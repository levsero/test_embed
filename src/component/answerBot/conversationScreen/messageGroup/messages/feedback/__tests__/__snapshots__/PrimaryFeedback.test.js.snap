// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the expected classes 1`] = `
.c1.c1.c1 {
  background-color: transparent !important;
  color: #1F73B7 !important;
  border-color: #1F73B7 !important;
  font-size: 1.0714285714285714rem !important;
  padding-right: 1.7142857142857142rem !important;
  padding-left: 1.7142857142857142rem !important;
  min-width: 0 !important;
  padding: 0.5714285714285714rem 1rem !important;
  border-radius: 1.1428571428571428rem !important;
  border-width: 0.07142857142857142rem !important;
  white-space: normal !important;
  line-height: 1.3 !important;
  height: auto !important;
}

.c1.c1.c1:not([disabled]):hover,
.c1.c1.c1:not([disabled]):active,
.c1.c1.c1:not([disabled]):focus {
  color: #FFFFFF !important;
  background-color: #1F73B7 !important;
  border-color: #1F73B7 !important;
}

.c0 {
  margin-bottom: 0.5rem;
}

<div>
  <div
    class="c0"
  >
    <div
      class="container buttonRight "
    >
      <button
        class="c1 index__c-btn___3y2Qe "
        data-garden-container-id="keyboardfocus"
        data-garden-container-version="0.2.6"
        data-garden-id="buttons.button"
        data-garden-version="7.0.2"
        data-testid="pill-button"
        tabindex="0"
        type="button"
      >
        Yes
      </button>
      <button
        class="c1 index__c-btn___3y2Qe "
        data-garden-container-id="keyboardfocus"
        data-garden-container-version="0.2.6"
        data-garden-id="buttons.button"
        data-garden-version="7.0.2"
        data-testid="pill-button"
        tabindex="0"
        type="button"
      >
        No, I need help
      </button>
    </div>
  </div>
</div>
`;
